packages/server-wallet/src/protocols/__test__/defunder.test.ts

```diff
@@ -37,6 +62,59 @@ beforeEach(async () => {
   );
 });
 
+describe('Collaborative transaction submitter', () => {
+  it('_shouldSubmitCollaborativeTx is correct for participant 0', async () => {
+    await testChannel.insertInto(store, {
+      participant: 0,
+      states: [3, 4],
+    });
+    const channel = await Channel.forId(testChannel.channelId, testKnex);
+
+    const objective = {
+      type: 'CloseChannel',
+      data: {txSubmitterOder: [0, 1]},
+    } as DBCloseChannelObjective;
+    expect(Defunder._shouldSubmitCollaborativeTx(channel, objective)).toEqual(true);
+
+    const objective2 = {
+      type: 'CloseChannel',
+      data: {txSubmitterOder: [1, 0]},
+    } as DBCloseChannelObjective;
+    expect(Defunder._shouldSubmitCollaborativeTx(channel, objective2)).toEqual(false);
+
+    const objective3 = ({
+      type: 'CloseChannel',
+      data: {txSubmitterOder: []},
+    } as unknown) as DBCloseChannelObjective;
+    expect(Defunder._shouldSubmitCollaborativeTx(channel, objective3)).toEqual(true);
```

Comment:
There's a lot of repeated code here. Is it worth defining a simple helper?
```
function closeObjective(txSubmitterOrder: number[]): DBCloseChannelObjective {
  return {
      type: 'CloseChannel',
      data: {txSubmitterOder},
    } as DBCloseChannelObjective;
}
```

I see the pattern repeated below as well.

Reply:
Good call 03dfd2d6c